src/Diagnostics/CodeAnalysis/Core/CodeAnalysisDiagnosticsResources.resx

```diff
@@ -174,6 +174,26 @@
   <data name="InvalidSyntaxKindTypeArgumentDescription" xml:space="preserve">
     <value>DiagnosticAnalyzer's language-specific Register methods, such as RegisterSyntaxNodeAction, RegisterCodeBlockStartAction and RegisterCodeBlockEndAction, expect a language-specific 'SyntaxKind' type argument for it's '{0}' type parameter. Otherwise, the registered analyzer action can never be invoked during analysis.</value>
   </data>
+  <data name="StartActionWithOnlyEndActionTitle" xml:space="preserve">
+    <value>Start action has no registered non-end actions.</value>
+  </data>
+  <data name="StartActionWithNoRegisteredActionsMessage" xml:space="preserve">
+    <value>You did not register any analyzer action on '{0}'. Move all the action registrations in '{1}' that depend on this start action onto '{0}'.</value>
+  </data>
+  <data name="StartActionWithNoRegisteredActionsTitle" xml:space="preserve">
+    <value>Start action has no registered actions.</value>
+  </data>
+  <data name="StartActionWithOnlyEndActionMessage" xml:space="preserve">
+    <value>You did not register any analyzer action on '{0}', except for a '{1}'. Either replace this start/end action pair with a '{2}' or move all the action registrations in '{3}' that depend on this start action onto '{0}'.</value>
+  </data>
+  <data name="StatefulAnalyzerRegisterActionsDescription" xml:space="preserve">
+    <value>An analyzer start action enables performing stateful analysis over a given code unit, such as a code block, compilation, etc in a performant way. However, incorrect design for such analyzers can lead to memory leaks and non-performant analyzer execution. Use the following guidelines for writing such analyzers:
+1. Define a new scope for the registered start action, possibly with a private nested type for analyzing each code unit.
+2. If required, define and intialize state object(s) in the start action within this scope.
+3. Register intermediate non-end actions in the start action, that depend upon and/or modify this state. You must register at least one such action. If there are no dependent intermediate actions for a start action, then the start action should be replaced by a corresponding non-start action. For example, a CodeBlockStartAction with no registered actions or only a registered CodeBlockEndAction should be replaced with a CodeBlockAction.
```

Comment:
How about something like:

Register at least one non-end action that refers to this state in the start action. If no such action is necessary, consider replacing the start action with a non-start action.

?